### ISSUE TITLE
MessageConnection onRequest and onNotification overloads that accept method name prefix

### ISSUE THREAD

**Issue by AArnott** [microsoft/vscode-languageserver-node#657]
To support an extension to JSON-RPC that I'm developing where we may marshal/proxy objects instead of serialize them so that methods on those objects may be invoked, I need to be able to add handlers given a method name prefix:

```ts
        function registerRequestAndNotification(connection: MessageConnection, method: any) {
            const methodPrefix = "$/invokeProxy/"
            connection.onRequestWithPrefix(methodPrefix, (...args: []) => method.apply(rpcTarget, args));
            connection.onNotificationWithPrefix(methodPrefix, (...args: []) => method.apply(rpcTarget, args));
        }
```

Can support for this be added to the `vscode-jsonrpc` package?

**Comment by AArnott** [comment]
This request can be Won't Fixed if #661 can be fulfilled.

**Comment by dbaeumer** [comment]
I fixed #661 since it is a more general implementation.

Please ping if this is still necessary in another context.
